ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxLimits.hs

```diff
@@ -43,46 +47,58 @@ import           Ouroboros.Consensus.Ticked (Ticked (..))
-  type Measure blk
+class BoundedMeasure (TxMeasure blk) => TxLimits blk where
+  type TxMeasure blk
 
-  txMeasure    :: Validated (GenTx blk) -> Measure blk
-  maxCapacity  :: Ticked (LedgerState blk) -> Measure blk
-  pointwiseMin :: Measure blk -> Measure blk -> Measure blk
+  -- | What is the measure an individual tx?
+  txMeasure        :: Validated (GenTx blk)    -> TxMeasure blk
 
--- | Is every part of the first measure less-than-or-equal-to the corresponding
--- part of the second measure?
---
--- See <https://en.wikipedia.org/wiki/Product_order>.
-lessEq :: forall blk. TxLimits blk => Measure blk -> Measure blk -> Bool
-lessEq x y = x == pointwiseMin @blk x y
+  -- | What is the allowed capacity for txs in an individual block?
+  txsBlockCapacity :: Ticked (LedgerState blk) -> TxMeasure blk
 
-newtype ByteSize = ByteSize { unByteSize :: Word32 }
-  deriving stock (Show, Eq, Ord)
+-- | Is every component of the first value less-than-or-equal-to the
+-- corresponding component of the second value?
+(<=) :: Measure a => a -> a -> Bool
+(<=) = (Measure.<=)
 
-instance Semigroup ByteSize where
-  (ByteSize bs1) <> (ByteSize bs2) = ByteSize $ bs1 + bs2
+{-------------------------------------------------------------------------------
+  ByteSize
+-------------------------------------------------------------------------------}
 
-instance Monoid ByteSize where
-  mempty = ByteSize 0
+newtype ByteSize = ByteSize { unByteSize :: Word32 }
+  deriving stock (Show)
+  deriving newtype (Eq, Ord)
+  deriving newtype (BoundedMeasure, Measure)
+
+{-------------------------------------------------------------------------------
+  Overrides
+-------------------------------------------------------------------------------}
 
 -- | How to override the limits set by the ledger state
 --
--- The forge logic must use the 'pointwiseMin'imum of the limits from ledger
--- state and from the result of this override.
-newtype Overrides blk = Overrides (Measure blk -> Measure blk)
+-- The forge logic must use the @Measure.min@ of the limits from ledger state
+-- and this override. More concretely: for each component of @'TxMeasure' blk@
+-- it will use the minimum of the ledger's limits and this override.
+newtype Overrides blk = Overrides {getOverrides :: TxMeasure blk}
+
+instance TxLimits blk => Monoid (Overrides blk) where
+  mempty = Overrides Measure.maxBound
+
+instance TxLimits blk => Semigroup (Overrides blk) where
+  (<>) = coerce $ Measure.min @(TxMeasure blk)
 
--- | Do not alter the limits set by the ledger state
-noOverrides :: Overrides blk
-noOverrides = Overrides id
+-- | @'applyOverrides' 'noOverrides' m = m@
+noOverrides :: TxLimits blk => Overrides blk
+noOverrides = mkOverrides id
 
-mkOverrides :: (Measure blk -> Measure blk) -> Overrides blk
-mkOverrides = Overrides
+-- | Define an override as some update of @'getOverrides' 'noOverrides'@
+mkOverrides :: TxLimits blk => (TxMeasure blk -> TxMeasure blk) -> Overrides blk
```

Comment:
Remove the function, change `noOverrides` to `noOverridesMeasure :: BoundedMeasure a => a`.